src/java/org/apache/cassandra/index/sai/disk/v1/V1OnDiskFormat.java

```diff
@@ -0,0 +1,167 @@
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.disk.v1;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.util.EnumSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.cassandra.db.compaction.OperationType;
+import org.apache.cassandra.db.lifecycle.LifecycleNewTracker;
+import org.apache.cassandra.index.sai.IndexContext;
+import org.apache.cassandra.index.sai.StorageAttachedIndex;
+import org.apache.cassandra.index.sai.disk.ColumnIndexWriter;
+import org.apache.cassandra.index.sai.disk.PerSSTableComponentsWriter;
+import org.apache.cassandra.index.sai.disk.format.IndexComponent;
+import org.apache.cassandra.index.sai.disk.format.IndexDescriptor;
+import org.apache.cassandra.index.sai.disk.format.OnDiskFormat;
+import org.apache.cassandra.index.sai.disk.v1.writers.MemtableIndexWriter;
+import org.apache.cassandra.index.sai.disk.v1.writers.SSTableComponentsWriter;
+import org.apache.cassandra.index.sai.disk.v1.writers.SSTableIndexWriter;
+import org.apache.cassandra.index.sai.memory.RowMapping;
+import org.apache.cassandra.index.sai.utils.NamedMemoryLimiter;
+import org.apache.cassandra.index.sai.utils.SAICodecUtils;
+import org.apache.cassandra.schema.CompressionParams;
+import org.apache.lucene.store.IndexInput;
+
+import static org.apache.cassandra.index.sai.StorageAttachedIndex.SEGMENT_BUILD_MEMORY_LIMITER;
+import static org.apache.cassandra.utils.FBUtilities.prettyPrintMemory;
+
+public class V1OnDiskFormat implements OnDiskFormat
+{
+    private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    private static final EnumSet<IndexComponent.Type> LITERAL_COMPONENT_TYPES = EnumSet.of(IndexComponent.Type.COLUMN_COMPLETION_MARKER,
+                                                                                           IndexComponent.Type.META,
+                                                                                           IndexComponent.Type.TERMS_DATA,
+                                                                                           IndexComponent.Type.POSTING_LISTS);
+    private static final EnumSet<IndexComponent.Type> NUMERIC_COMPONENT_TYPES = EnumSet.of(IndexComponent.Type.COLUMN_COMPLETION_MARKER,
+                                                                                           IndexComponent.Type.META,
+                                                                                           IndexComponent.Type.KD_TREE,
+                                                                                           IndexComponent.Type.KD_TREE_POSTING_LISTS);
+
+    public static final V1OnDiskFormat instance = new V1OnDiskFormat();
+
+    private V1OnDiskFormat()
+    {}
+
+    @Override
+    public boolean isGroupIndexComplete(IndexDescriptor indexDescriptor)
```

Comment:
This method looks like a getter, but has a side-effect of registering the sstable.

Reply:
The `registerSSTable` method is misnamed. It's purpose is to update the file cache in the `IndexDescriptor` before checking for file existence. I put this there as a safety feature during development. I'm hoping that, once tested, we can get rid of this additional step.

I have been trying to unravel a lot of spaghetti code around file creation and this was there because of it.